### PR TITLE
Refactor Ballerina String API.

### PR DESCRIPTION
**Pull request by djkevincr**

### CODE REVIEW

File: modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal [hunk 1]
```diff
@@ -5,3 +5,98 @@ package ballerina.builtin;
 @Param { value:"anotherString: The string to be compared" }
 @Return { value:"boolean: True if the strings are equal; false otherwise" }
 public native function <string mainString> equalsIgnoreCase (string anotherString) (boolean);
+
+@Description { value:"Returns a string with all the characters converted to uppercase"}
+@Param { value:"s: The original string argument" }
+@Return { value:"string: The string converted to uppercase" }
+public native function <string s> toUpperCase () (string);
```
Comment (anupama-pathirage) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal]: Shall we either use <string mainString> or <string s> throughout the whole file to maintain the consistency.

File: modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal [hunk 2]
```diff
@@ -5,3 +5,98 @@ package ballerina.builtin;
...
+public native function <string mainString> replace (string replacePattern, string replaceWith) (string);
+
+@Description { value:"Splits the string with the given regular expression to produce a string array."}
+@Param { value:"mainString: The original string argument" }
+@Param { value:"regex: The regex to split the string" }
+@Return { value:"string[]: The split string array" }
+public native function <string mainString> split (string regex) (string[]);
+
+@Description { value:"Converts string to a BLOB"}
+@Param { value:"b: string value to be converted" }
```
Comment (anupama-pathirage) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal]: What is "b" here?

File: modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal [hunk 3]
```diff
@@ -5,3 +5,98 @@ package ballerina.builtin;
...
+@Param { value:"encoding: Encoding to used in the conversion" }
+@Return { value:"string: BLOB representation of the given String" }
```
Comment (anupama-pathirage) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal]: string should be changed to blob

File: modules/ballerina-builtin/src/main/ballerina/ballerina/lang/strings/natives.bal [hunk 4]
```diff
@@ -2,102 +2,7 @@ package ballerina.lang.strings;
...
-@doc:Return { value:"string[]: The split string array" }
-public native function split (string mainString, string regex) (string[]);
-
-@doc:Description { value:"Converts string to a BLOB"}
-@doc:Param { value:"b: string value to be converted" }
-@doc:Param { value:"encoding: Encoding to used in the conversion" }
-@doc:Return { value:"string: BLOB representation of the given String" }
-public native function toBlob (string s, string encoding) (blob);
\ No newline at end of file
+public native function valueOf (any value) (string);
```
Comment (anupama-pathirage) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/lang/strings/natives.bal]: What is the reason for keeping the valueOf function in ballerina.lang.Strings package?

File: modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal (hunk 3 above)
Comment (djkevincr) [reply modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal]: +1

File: modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal (hunk 2 above)
Comment (djkevincr) [reply modules/ballerina-builtin/src/main/ballerina/ballerina/builtin/stringlib.bal]: +1

File: modules/ballerina-builtin/src/main/ballerina/ballerina/lang/strings/natives.bal (hunk 4 above)
Comment (djkevincr) [reply modules/ballerina-builtin/src/main/ballerina/ballerina/lang/strings/natives.bal]: This is needed to removed from any type.

**Pull request merged**